artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumIntegration/Mappers/BurnDown/SprintStatsResponseMapper.cs

```diff
@@ -0,0 +1,77 @@
+using Artificial.Scrum.Master.ScrumIntegration.Features.Burndown;
+using System.Text.RegularExpressions;
+
+namespace Artificial.Scrum.Master.ScrumIntegration.Mappers.BurnDown;
+
+internal interface ISprintStatsResponseMapper
+{
+    GetSprintStatsResponse MapSprintStatsResponse(SprintStats sprintStats, SprintData sprintData);
+}
+
+internal class SprintStatsResponseMapper : ISprintStatsResponseMapper
+{
+    private static readonly Regex RoleRegex = new(@"\D");
+
+    public GetSprintStatsResponse MapSprintStatsResponse(SprintStats sprintStats, SprintData sprintData)
+    {
+        var sprintDayStats = sprintStats.Days?.Select(day => new SprintDayStats(
+            Day: day.Day ?? string.Empty,
+            Name: day.Name,
+            OpenPoints: day.OpenPoints,
+            OptimalPoints: day.OptimalPoints
+        )).ToList();
+
+        var totalRolePoints = MapTotalRolePoints(sprintStats.TotalPoints, sprintData);
+        var completedPoints = sprintStats.CompletedPoints?.Sum() ?? 0;
+
+        return new GetSprintStatsResponse(
+            SprintName: sprintStats.Name ?? string.Empty,
+            EstimatedStart: sprintStats.EstimatedStart ?? string.Empty,
+            EstimatedFinish: sprintStats.EstimatedFinish ?? string.Empty,
+            TotalUserStories: sprintStats.TotalUserStories,
+            CompletedUserStories: sprintStats.CompletedUserStories,
+            TotalTasks: sprintStats.TotalTasks,
+            CompletedTasks: sprintStats.CompletedTasks,
+            SprintDayStats: sprintDayStats ?? [],
+            TotalRolePoints: totalRolePoints,
+            CompletedRolePoints: completedPoints
+        );
+    }
+
+    private static List<KeyValuePair<string, double>> MapTotalRolePoints(
```

Comment:
Osobiście to bym widział jako osobną klasę z testami jednostkowymi czy ten regex napewno działa bo szczerze jestem ułomny w tych sprawach więc musze uwierzyć na słowo xD